### PR TITLE
Updates to support Catchment-CN.4.5 in addition to Catchment-CN.4.0

### PR DESCRIPTION
**Pull request by gmao-jkolassa**
This PR is linked to GEOSgcm_GridComp PR [368](https://github.com/GEOS-ESM/GEOSgcm_GridComp/pull/368) and GEOSldas PR [326](https://github.com/GEOS-ESM/GEOSldas/pull/326). The objective is to implement modification required to support both Catchment-CN.4.0 and Catchment-CN.4.5. Specific files affected are:

**AGCM.rc.tmpl:**
- added lightning frequency climatology file required by Catchment-CN.4.5

**gcm_run.j:**
- added lightning frequency climatology file (required by Catchment-CN.4.5) for GCMRUN_CATCHCN

cc: @gmao-rreichle , @biljanaorescanin , @rdkoster , @smahanam

### CODE REVIEW

**Comment by biljanaorescanin** [comment]
Label DNA until conflict with linked PR in GEOSgcm_GridComp (PR 368) is resolved.

File: gcm_setup [hunk 1]
```diff
@@ -1146,20 +1146,27 @@ endif
...
         echo
     endif
     if( $LSM_CHOICE == 1 ) set HIST_CATCHCN = "#DELETE"
     if( $LSM_CHOICE == 2 ) set HIST_CATCHCN = ""
+    if( $LSM_CHOICE == 3 ) set HIST_CATCHCN = ""
     if( $LSM_CHOICE == 1 ) set GCMRUN_CATCHCN = "#DELETE"
     if( $LSM_CHOICE == 2 ) set GCMRUN_CATCHCN = ""
+    if( $LSM_CHOICE == 3 ) set GCMRUN_CATCHCN = ""
+    if( $LSM_CHOICE == 2 ) then
+      echo "IMPORTANT: please set LAND_PARAMS: to CN_CLM40 in RC/GEOS_SurfaceGridComp.rc in the experiment directory."
```
Comment (gmao-rreichle) [inline gcm_setup:1166]: @gmao-jkolassa 
The help message here ("CN_CLM40") is not consistent with the LAND_PARAMS value ("CN_CLM4").   See my earlier GEOSldas PR comment about the inconsistency in the model naming syntax:
https://github.com/GEOS-ESM/GEOSldas/pull/326#discussion_r661817271

File: AGCM.rc.tmpl [hunk 2]
```diff
@@ -522,9 +523,17 @@ CATCH_INTERNAL_RESTART_FILE:            catch_internal_rst
...
-CATCHCN_INTERNAL_CHECKPOINT_TYPE:         @CHECKPOINT_TYPE
+CATCHCNCLM40_INTERNAL_RESTART_FILE:     catchcn_internal_rst
+CATCHCNCLM40_INTERNAL_CHECKPOINT_FILE:  catchcn_internal_checkpoint
+CATCHCNCLM40_INTERNAL_CHECKPOINT_TYPE:  @CHECKPOINT_TYPE
+
+CATCHCNCLM45_INTERNAL_RESTART_FILE:     catchcn_internal_rst
+CATCHCNCLM45_INTERNAL_CHECKPOINT_FILE:  catchcn_internal_checkpoint
+CATCHCNCLM45_INTERNAL_CHECKPOINT_TYPE:  @CHECKPOINT_TYPE
+
+IRRIGATION_INTERNAL_RESTART_FILE:       irrigation_internal_rst
```
Comment (gmao-rreichle) [inline AGCM.rc.tmpl:534]: @gmao-jkolassa 
Why irrigation???

File: HISTORY.AGCM.rc.tmpl [hunk 3]
```diff
@@ -230,6 +230,7 @@ PC@HIST_IMx@HIST_JM-DC.LM: @AGCM_LM
 >>>HIST_CATCHCN<<<        'CNTOTC'   , 'SURFACE'  ,
 >>>HIST_CATCHCN<<<        'CNVEGC'   , 'SURFACE'  ,
 >>>HIST_CATCHCN<<<        'CNROOT'   , 'SURFACE'  ,
+>>>HIST_CATCHCN<<<        'CNFROOTC' , 'SURFACE'  ,
```
Comment (gmao-rreichle) [inline HISTORY.AGCM.rc.tmpl:233]: In the GEOSldas PR, the output of CNFROOTC is only for CatchCNCLM45.  Why make a distinction between CNCLM40 and CNCLM45 in GEOSldas but not in the GCM App?

Comment (smahanam) [reply HISTORY.AGCM.rc.tmpl:233]: This is correct. I checked my AGCM and GEOSldas test experiments and verified.

File: gcm_setup (hunk 1 above)
Comment (gmao-jkolassa) [reply gcm_setup:1166]: Addressed this by renaming LAMD_PARAMS for Catchment-CN4.0 to "CN_CLM40". The relevant commits are part of the PR for the GEOSgcm_GridComp [PR #368](https://github.com/GEOS-ESM/GEOSgcm_GridComp/pull/368)

File: AGCM.rc.tmpl (hunk 2 above)
Comment (gmao-jkolassa) [reply AGCM.rc.tmpl]: Removed irrigation restart file.

File: HISTORY.AGCM.rc.tmpl (hunk 3 above)
Comment (gmao-jkolassa) [reply HISTORY.AGCM.rc.tmpl:233]: This has been discussed and correctness of the implementation here has been verified.

**Comment by biljanaorescanin** [comment]
This PR is ready for review.

**Review by sdrabenh — APPROVED** [review]

**Pull request merged**